gems/aws-sdk-core/lib/aws-sdk-core/process_credentials.rb

```diff
@@ -0,0 +1,80 @@
+module Aws
+
+  # A credential provider that executes a given process and attempts
+  # to read its stdout to recieve a JSON payload containing the credentials
+  #
+  # Automatically handles refreshing credentials if an Expiration time is 
+  # provided in the credentials payload
+  #
+  #     credentials = Aws::ProcessCredentials.new(
+  #                   'echo \'{
+  #                             "Version": 1,
+  #                             "AccessKeyId": "AK_PROC1",
+  #                             "SecretAccessKey": "SECRET_AK_PROC1",
+  #                             "SessionToken": "TOKEN_PROC1"
+  #                            }\''
+  #                   ).credentials
+  #
+  #     ec2 = Aws::EC2::Client.new(credentials: credentials)
+  #
+  # More documentation on process based credentials can be found here:
+  # https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#sourcing-credentials-from-external-processes
+  class ProcessCredentials
+
+    include CredentialProvider
+    include RefreshingCredentials
+
+    # Creates a new ProcessCredentials object, which allows an
+    # external process to be used as a credential provider.
+    #
+    # @param [String] process Invocation string for process
+    # credentials provider. 
+    def initialize(process)
+      @process = process
+      @credentials = credentials_from_process(@process)
+    end
+
+    private
+    def credentials_from_process(proc_invocation)
+      begin
+        raw_out = `#{proc_invocation}`
```

Comment:
My last point is for us to take another pass on the process invocation choice. Backticks run this way are going to be unescaped, and while we're taking in the data from the credentials file, we may want to be cautious. Take a look at [Open3](http://ruby-doc.org/stdlib-2.3.0/libdoc/open3/rdoc/Open3.html) and let's see if `Open3.capture2` works equally well.